crates/core/src/table/mod.rs

```diff
@@ -474,6 +481,36 @@ impl Table {
     }
 }
 
+/// Formats a timestamp string to `yyyyMMddHHmmSSSSS` format.
+///
+/// The function attempts to parse the input timestamp string (`timestamp`) into a `DateTime` object.
+/// It first tries parsing the ISO 8601 format, and if that fails, it tries two other formats: `yyyyMMddHHmmSSSSS`
+/// and `yyyyMMddHHmmSS`. If none of the formats match, it returns the original input string.
+///
+/// # Arguments
+/// - `timestamp`: The input timestamp string to be formatted.
+///
+/// # Returns
+/// A string formatted as `yyyyMMddHHmmSSSSS`. If the input cannot be parsed, the original string is returned.
+fn format_timestamp(timestamp: &str) -> String {
```

Comment:
i'd expect this returning a Result - if users pass a timestamp str that can't be converted to hudi timeline format properly, then the flow should early return.

The user-provided timestamp should also conform to timezone config of the table - this is by default utc; but it could be configured as `local`.

Reply:
I have added an is_utc parameter to this function in the new commit to mark whether the utc time zone is used, and changed the return value to `Result`, it will return a CoreError::Timeline when parsing errors occur.